src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/SignatureBuilder.cs

```diff
@@ -519,22 +519,38 @@ public void EmitMethodRefToken(ModuleToken memberRefToken)
 
         public void EmitFieldSignature(FieldDesc field, SignatureContext context)
         {
+            uint fieldSigFlags = 0;
+            TypeDesc canonOwnerType = field.OwningType.ConvertToCanonForm(CanonicalFormKind.Specific);
+            TypeDesc ownerType = null;
+            if (canonOwnerType.IsCanonicalSubtype(CanonicalFormKind.Specific))
+            {
+                ownerType = field.OwningType;
+                fieldSigFlags |= (uint)ReadyToRunFieldSigFlags.READYTORUN_FIELD_SIG_OwnerType;
+
+                // Convert field to canonical form as this is what the field - module token lookup stores
+                field = field.Context.GetFieldForInstantiatedType(field, (InstantiatedType)canonOwnerType);
```

Comment:
```suggestion
                field = field.Context.GetFieldForInstantiatedType(field.GetTypicalFieldDefinition(), (InstantiatedType)canonOwnerType);
```

Reply:
I see, sorry about my overlooking, it seems to be working fine now. Thank you!